tests/test_font.py

```diff
@@ -22,4 +22,18 @@ def test_font2():
     """Old and new length computation must be the same."""
     font = fitz.Font("helv")
     text = "PyMuPDF"
-    assert font.text_length(text) == fitz.get_text_length(text)
\ No newline at end of file
+    assert font.text_length(text) == fitz.get_text_length(text)
+
+
+def test_fontname():
+    """Assert a valid PDF fontname."""
+    doc = fitz.open()
+    page = doc.new_page()
+    font = fitz.Font("helv")
+    assert page.insert_font()
+    detected = False  # preset indicator
+    try:
+        page.insert_font(fontname="illegal/char", fontbuffer=font.buffer)
+    except ValueError:
+        detected = True  # illegal fontname detected
```

Comment:
Might be good to check the text of the exception? Maybe not worth it though, will leave it to you.